packages/beacon-node/src/api/impl/beacon/blocks/utils.ts

```diff
@@ -22,50 +23,53 @@ export function toBeaconHeaderResponse(
   };
 }
 
-export async function resolveBlockId(
-  chain: IBeaconChain,
-  blockId: routes.beacon.BlockId
-): Promise<{block: SignedBeaconBlock; executionOptimistic: boolean; finalized: boolean}> {
-  const res = await resolveBlockIdOrNull(chain, blockId);
-  if (!res) {
-    throw new ApiError(404, `No block found for id '${blockId}'`);
-  }
-
-  return res;
-}
-
-async function resolveBlockIdOrNull(
-  chain: IBeaconChain,
-  blockId: routes.beacon.BlockId
-): Promise<{block: SignedBeaconBlock; executionOptimistic: boolean; finalized: boolean} | null> {
+export function resolveBlockId(forkChoice: IForkChoice, blockId: routes.beacon.BlockId): string | number {
```

Comment:
could reuse type from api here

```suggestion
export function resolveBlockId(forkChoice: IForkChoice, blockId: routes.beacon.BlockId): routes.beacon.BlockId {
```

Reply:
hah, I'd prefer not to do that. Structurally the same, semantically not the same

Reply:
ah nvm this comment, block id also has `"head" | "genesis" | "finalized" | "justified"`, so yeah, it's not the same